### PR TITLE
Improved wires support

### PR DESCRIPTION
**Pull request by trbromley**
*Description of changes:* In PL-core, it is possible for `wires` to be a list of strings and/or numbers:
```python
import pennylane as qml

dev = qml.device("default.qubit", wires=["a", -1])

@qml.qnode(dev)
def f(x):
    qml.RX(x, wires="a")
    qml.RZ(x, wires=-1)
    return qml.expval(qml.PauliZ("a") @ qml.PauliZ(-1))

f(0.3)
```
This PR extends support to the Braket device.

*Testing done:* A `test_wires` test has been added. Furthermore, we have added the `braket.local.qubit` device to the [PL testing matrix](https://github.com/PennyLaneAI/plugin-test-matrix#testing-matrix). Here, the device is tested against the [standard device test suite](https://github.com/PennyLaneAI/pennylane/tree/master/pennylane/devices/tests). It was the failures there within the `test_wires.py` file that made us notice this.

### CODE REVIEW

File: src/braket/pennylane_plugin/braket_device.py [hunk 1]
```diff
@@ -124,7 +126,8 @@ def _pl_to_braket_circuit(self, circuit, **run_kwargs):
             **run_kwargs,
         )
         for observable in circuit.observables:
-            braket_circuit.add_result_type(translate_result_type(observable))
+            dev_wires = self.map_wires(observable.wires).tolist()
```
Comment (antalszava) [inline src/braket/pennylane_plugin/braket_device.py:129]: Just to make sure: `braket_circuit` would be awaiting consecutive wire ordering, correct? That's why we need to map the custom device wires using `map_wires`.

File: src/braket/pennylane_plugin/translation.py [hunk 2]
```diff
@@ -217,18 +217,18 @@ def _(zz: ZZ, parameters):
     return gates.ZZ(-phi) if zz.inverse else gates.ZZ(phi)
 
 
-def translate_result_type(observable: Observable) -> ResultType:
+def translate_result_type(observable: Observable, targets: List[int]) -> ResultType:
     """Translates a PennyLane ``Observable`` into the corresponding Braket ``ResultType``.
 
     Args:
         observable (Observable): The PennyLane ``Observable`` to translate
+        targets (List[int]): The target wires of the observable
```
Comment (antalszava) [inline src/braket/pennylane_plugin/translation.py:225]:
```suggestion
        targets (List[int]): The target wires of the observable using a consecutive integer wire ordering
```

**Review by antalszava — COMMENTED** [review]
Looks good to me!

File: src/braket/pennylane_plugin/translation.py (hunk 2 above)
Comment (trbromley) [reply src/braket/pennylane_plugin/translation.py:225]: Nice!

**Review by speller26 — APPROVED** [review]

**Pull request merged**